### PR TITLE
Added fix for #840

### PR DESCRIPTION
**Pull request by regentcid434**
## Description
A few sentences describing the overall goals of the pull request's commits.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/adazzle/react-data-grid/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)
#840 

**What is the new behavior?**
`getRenderedColumnCount` now accounts for the first visible column being partially offscreen and adjusts its calculation to only include to visible width of the column

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

**Other information**:

### CODE REVIEW

File: packages/react-data-grid/src/ColumnUtils.js [hunk 1]
```diff
@@ -43,5 +43,20 @@ module.exports = {
       value = column[property];
     }
     return value;
+  },
+
+  getWidthOfColumns: function getSize(columns, startIdx, endIdx) {
+    let totalWidth = 0;
+    if (Array.isArray(columns)) {
+      for (let idx = startIdx; idx <= endIdx; idx++) {
+        totalWidth += columns[idx].width;
```
Comment (jpdriver) [inline packages/react-data-grid/src/ColumnUtils.js]: I don't think we can rely on columns always having an explicit `width` here.

Can you add some unit tests -- where all columns have widths, only some have widths, and none have widths?

Comment (regentcid434) [reply packages/react-data-grid/src/ColumnUtils.js]: @jpdriver I've tested this locally (still need to make unit tests) and width always has a value. The library has a hard-coded minWidth of 80, so a column won't be smaller than that unless you specify one explicitly. Columns without explicit widths set act as if they were each `flex: 1`.

In the current implementation of `getRenderedColumnCount`, the column widths are already being accessed in the same way my new util function does. Each iteration, the column metrics for the given index are retrieved and the width is subtracted from `remainingWidth`.
```
getRenderedColumnCount(displayStart, width) {
    let remainingWidth = width && width > 0 ? width : this.props.columnMetrics.totalWidth;
    if (remainingWidth === 0) {
      remainingWidth = ReactDOM.findDOMNode(this).offsetWidth;
    }
    let columnIndex = displayStart;
    let columnCount = 0;
    while (remainingWidth > 0) {
      let column = ColumnUtils.getColumn(this.props.columnMetrics.columns, columnIndex);

      if (!column) {
        break;
      }

      columnCount++;
      columnIndex++;
      remainingWidth -= column.width;
    }
    return columnCount;
  }
```

**Comment by kevyu** [comment]
can this pr be merged?

**Comment by jrymer** [comment]
What is the status on this PR?

**Comment by regentcid434** [comment]
I've just been using my branch since it was a production issue that needed to be addressed, and I've not gotten around to any unit tests. They said that my calculation uses a width that might not have a value. However, the current implementation uses that same value. I'm just changing the math around it.
